ДЗ выполнено не полностью

TatyanaBogodvid opened the pull request:
Создан класс Recipe (рецепт);
Создан класс Ingredient (ингредиент);
Создан интерфейсы и классы сервисов с методами;

TatyanaBogodvid committed:
Создан класс Recipe (рецепт);
Создан класс Ingredient (ингредиент);
Создан интерфейсы и классы сервисов с методами;

TatyanaBogodvid committed:
Созданы контроллеры для создания и получения рецептов и ингредиентов?

TatyanaBogodvid committed:
Внесены изменения в методы контроллеров и сервисов

TatyanaBogodvid committed:
Добавлены методы редактирования рецептов и ингредиентов по id.
Добавлены методы удаления рецептов и ингредиентов по id.
Добавлены методы удаления всех рецептов и ингредиентов.
Добавлены методы получения списка всех рецептов и ингредиентов.

TatyanaBogodvid committed:
К проекту подключена библиотека commons-lang3.
К проекту подключена библиотека Swagger.
Добавлен сервис ValidateService.
Добавлены описание ко всем контроллерам и эндпоинтам.

kalyashov commented on src/main/java/com/example/recipes/controllers/IngredientController.java:
```diff
@@ -0,0 +1,112 @@
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@Tag(name = "IngrediemtController", description = "API для ингредиентов")
```
Опечатка - IngredientController

kalyashov commented on src/main/java/com/example/recipes/controllers/IngredientController.java:
```diff
@@ -0,0 +1,112 @@
+        return ResponseEntity.ok(ingredient);
+    }
+
+    @Operation(summary = "Получение всего списка игредиентов", description = "Получение всего списка рингредиентов")
```
в description опечатка

kalyashov commented on src/main/java/com/example/recipes/services/IngredientService.java:
```diff
@@ -0,0 +1,17 @@
+import com.example.recipes.model.Ingredient;
+
+public interface IngredientService {
+    public long addIngredient(Ingredient ingredient);
```
Это интерфейс, модификаторы доступа не нужны, немного поехало форматирование - между методами нужен 1 перенос строки

kalyashov commented on src/main/java/com/example/recipes/services/RecipeService.java:
```diff
@@ -0,0 +1,17 @@
+import com.example.recipes.model.Recipe;
+
+public interface RecipeService {
+    public long addRecipe(Recipe recipe);
```
модификаторы доступа можно убрать

TatyanaBogodvid committed:
Вопросы по ДЗ

kalyashov commented on src/main/java/com/example/recipes/services/impl/IngredientServiceImpl.java:
```diff
@@ -0,0 +1,98 @@
+
+@Service
+public class IngredientServiceImpl implements IngredientService {
+    private final FilesService filesService;
```
Используй здесь реализацию FilesService для ингредиентов, то есть FilesServiceIngImpl

kalyashov commented on src/main/java/com/example/recipes/services/impl/RecipeServiceImpl.java:
```diff
@@ -0,0 +1,101 @@
+@Service
+public class RecipeServiceImpl implements RecipeService {
+
+    private final FilesService filesService;
```
Используй здесь реализацию FilesService для рецептов, то есть FilesServiceRecImpl

TatyanaBogodvid committed:
В сервисе реализован метод сохранения данных в формате json в файле на жестком диске.
В сервисе реализован метод чтения данных с диска.
В сервисе с рецептами должен быть реализован метод с аннотацией @PostConstruct.
В проекте обработаны исключения: отладочная информация выводится в консоль e.printStackTrace().
Данные по рецептам и по ингредиентам сохраняются в разные файлы.

TatyanaBogodvid committed:
ДЗ 3.7 выполнено

TatyanaBogodvid committed:
ДЗ Дебаг

TatyanaBogodvid committed:
ДЗ Дебаг

TatyanaBogodvid committed:
ДЗ Дебаг выполнено
